File: Cap 10/10.8 - Recursividade ou Recorrencia/10.8.1 - Recursividade Simples/recsimp.php
<?php
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.8
 * 
 * Programa ..: FUNCAO COM RECURSIVDADE SIMPLES
 */

function fatorial($N) {
    if ($N == 0) {
        return 1;
    } else {
        return $N * fatorial($N - 1);
    }
}

echo "CALCULO DE FATORIAL\n";
echo "RECURSIVIDADE SIMPLES\n";
echo "\n";

echo "Qual fatorial (1-20): ";
$LIMITE = intval(fgets(STDIN));

echo "Fatorial = " . fatorial($LIMITE) . "\n";

?>

File: Cap 10/10.8 - Recursividade ou Recorrencia/10.8.1 - Recursividade Simples/recsimp.rb
=begin
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.8
 * 
 * Programa ..: FUNCAO COM RECURSIVDADE SIMPLES
=end

def fatorial(n)
  if n == 0
    1
  else
    n * fatorial(n - 1)
  end
end

limite = 0

puts 'CALCULO DE FATORIAL'
puts 'RECURSIVIDADE SIMPLES'
puts

print 'Qual fatorial (1-9358): '
limite = gets.chomp.to_i

puts 'Fatorial =', fatorial(limite)
